Studies/python/BayesianOptimizationCustom.py

```diff
@@ -174,7 +199,11 @@ def maximize(self, n_iter, acq='ucb', kappa=2.576, xi=0.0):
         for key, values in self.params_typed_range.items():
             new_target_point = copy.deepcopy(p_target_max)
             for v in values['range']:
-                new_target_point[key] = v
+                # the variable on which it depends has a value of zero
+                if 'depends_on' in self.params_typed_range[key] and new_target_point[self.params_typed_range[key]['depends_on'] ] == 0:
```

Comment:
if this condition is satisfied it makes no sense to make an inner loop. should do this check in the outer loop, and do continue.